### ISSUE TITLE
preferCanvas problem with panes

### ISSUE THREAD

**Issue by mrit64** [Leaflet/Leaflet#5190]
## How to reproduce

- Leaflet version I'm using: 1.0.2
- Browser (with version) I'm using: Chrome and FF
- OS/Platform (with version) I'm using: Windows 10

## What behaviour I'm expecting and which behaviour I'm seeing
I have crated a map with preferCanvas option (or renderer: L.canvas()). It's working well without panes. When panes are associated with layers, SVG renderer is used despite the preferCanvas option.
Maybe because of the following instruction in L.Map._getPaneRenderer  function :

		if (renderer === undefined) {
			renderer = (L.SVG && L.svg({pane: name})) || (L.Canvas && L.canvas({pane: name}));

...which favour SVG renderer in all cases ?

Best regards

**Comment by perliedman** [comment]
@mrit64 could you provide a running example on http://playground-leaflet.rhcloud.com/ or similar?

**Comment by mrit64** [comment]
Here : 
https://jsfiddle.net/mrit64/wrq77j6e/

**Comment by Lax-Phorell** [comment]
I have the same problem, but i think it is connected with
```javascript
getPaneRenderer: function (name) {
    if (name === 'overlayPane' || name === undefined) {
        return false;
    }
...
```
I think by default, Leaflet uses the overlayPane when rendering to canvas. If you use a different name for your pane, then
```javascript
renderer = (L.SVG && L.svg({pane: name})) || (L.Canvas && L.canvas({pane: name}));
```
kicks in and you end up with a SVG renderer and therefore `this.options.preferCanvas` isn't even checked.

**Comment by IvanSanchez** [comment]
Closed by #6019
